webpack/assets/javascripts/react_app/components/ForemanModal/__tests__/ForemanModal.integration.test.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import ForemanModal, { reducers } from '../index';
+import ForemanModalHeader from '../subcomponents/ForemanModalHeader';
+import ForemanModalFooter from '../subcomponents/ForemanModalFooter';
+import IntegrationTestHelper from '../../../common/IntegrationTestHelper';
+
+import { setModalOpen, setModalClosed, addModal } from '../ForemanModalActions';
+
+// This file is for integration tests of the Redux-connected ForemanModal component
+
+describe('ForemanModal - integration tests', () => {
+  it('should add, open, and close modals as directed by Redux actions', () => {
+    const integrationTestHelper = new IntegrationTestHelper(reducers);
+
+    integrationTestHelper.takeStoreSnapshot('initial state');
+
+    integrationTestHelper.store.dispatch(addModal({ id: 'modal1' }));
+    integrationTestHelper.store.dispatch(addModal({ id: 'modal2' }));
+    integrationTestHelper.store.dispatch(addModal({ id: 'modal3' }));
+
+    const modal1 = integrationTestHelper.mount(
```

Comment:
I'm wondering, instead of manually dispatching `addModal` before mounting. Would it make sense if `ForemanModal` will automatically call it `onMount` and will clean the state `onUnmount`?

Reply:
Ah, so have `ForemanModal` itself handle dispatching the `addModal` action for you.  nice idea!

Since ForemanModal is a function component and not a class, I will do it with `useEffect` rather than refactoring to a class component just to use `componentDidMount` and `componentWillUnmount`.

Reply:
Yes! and IMO the `useEffect` should be used inside the `ConnectedForemanModal`

Reply:
@sharvit Updated